src/interpret.c

```diff
@@ -1810,22 +1810,46 @@ class Interpreter : public Visitor
     #endif
         if (istate->start == s)
             istate->start = NULL;
-        if (istate->start)
+
+        Expression *ex = NULL;
+        // Jump into/out from finalbody is disabled in semantic analysis,
+        // so we only need to care about jump into the body. Jump within finalbody will be handled
+        // by the ScopeStatement == finalbody.
+        bool labelSeeking = istate->start != NULL;
+        ex = interpret(s->body, istate);
+        if (labelSeeking && istate->start != NULL)
         {
-            Expression *e = NULL;
-            e = interpret(s->body, istate);
-            // Jump into/out from finalbody is disabled in semantic analysis.
-            // and jump inside will be handled by the ScopeStatement == finalbody.
-            result = e;
+            // We were looking for a label, but did not find it, don't interpret the finalbody.
+            result = ex;
             return;
         }
-
-        Expression *ex = interpret(s->body, istate);
```

Comment:
These line changes (setting `labelSeeking` etc.) are unrelated with the bugfix. The `if (istate->start)` style is consistent with other interpreter parts. Please don't break consistency.

However, I don't deny the changes for code readability. If you think current code is not well understandable, please open a new PR for refactoring.